### ISSUE TITLE
RFC: Drop the @Powertools prefix from the annotations

### ISSUE THREAD

**Issue by dgomesbr** [awslabs/aws-lambda-powertools-java#68]
**Is your feature request related to a problem? Please describe.**

Java developers are used to multiple imports and naming conventions. `@Logging @Metric/Timed @Trace` and so on. By not using Powertools prefix it would adhere to style of other frameworks.

**Describe the solution you'd like**

Use shorter annotation names, as we already using this in a Lambda fashion where would simply make sense that we're importing the Powertools annotation. Java differently from other languages, has sorted this out thought import organization within IDEs.

**Describe alternatives you've considered**

Either continue with Prefixed Annotations, or simply abide to other more common names, without the prefix.

**Additional context**

1. https://docs.newrelic.com/docs/apm/transactions/transaction-traces/introduction-transaction-traces
2. https://micrometer.io/docs/concepts#_the_timed_annotation

**Comment by jeromevdl** [comment]
+1

We write @Test, not @JUnitTest, @Bean, not @SpringBean, ... the name of a class / method / attribute must refer to the problem it tries to solve and not contain anything else. "Powertools" is in the package and that's enough.

Furthermore, tracing & logging are not consistent: PowerTracer vs PowertoolsLogger ==> Tracer & Logger would be enough (and more consistent with the python version).

I can take this one if you want...

**Comment by jeromevdl** [comment]
Several thoughts we have about naming...

Using the python names (`Tracer`, `Logger`), we face a conflict with the Log4j Logger, which would result in qualified import. And it is generally painful to have qualified import (either `org.apache.logging.log4j.Logger`or `software.amazon.lambda.logging.Logger`).

A solution can be to prefix those names with 'Lambda' to stay consistent:
`LambdaLogger`, `LambdaTracer` and same for the annotations: `@LambdaLogging`and `@LambdaTracing`
It is part of our tenet to provide this library for Lambda only so why not.

Thinking about the future and the Metrics module, we could also imagine: `Logs`, `Traces`, `Metrics` (and keep the `@Logging` and `@Tracing` annotations). Then, does the following make sense: `Logs.appendKey()`, `Traces.putMetadata()`?

We can also have other alternatives to remediate the Logger problem: `EnhancedLogger` or `StructuredLogger`

We must be consistent between modules, as much as possible also with the python library, but also respect the java standards and best practices.

My preference:
`LambdaLogger`, `LambdaTracer`, `LambdaMetrics` (`LambdaMonitor`/`LambdaMeter` or an equivalent would be more logic...), `@Logging`, `@Tracing`, `@MetricsLogging`(not sure for the Lambda prefix on annotations...)

What is your preference? (or better idea)

**Comment by dgomesbr** [comment]
> Using the python names (Tracer, Logger), we face a conflict with the Log4j Logger, which would result in qualified import. And it is generally painful to have qualified import (either org.apache.logging.log4j.Logger or software.amazon.lambda.logging.Logger).

This is a non-issue in Java land, IDE's will solve the context, and having log4j.Logger as a qualified import would not hurt. Also, this yet another discussion, why @Logger also provides a `logger` instance to the method/class? Akin to what Lombok does;

Using substantive rather than the verb would also make more sense: `@Logger`, `@Trace`, `@Timed/@Metric` and would keep the consistency.

I dislike the `Lambda*` prefix as redundant and could've easily being dropped because of the context (we're at a lambda nonetheless).

Thoughts?

**Comment by jeromevdl** [comment]
I agree the shorter the better. Ideally it would be: `Logger`, `Tracer` and ...?  and `@Log`, `@Trace`, `@LogMetrics` 
But having two Loggers will definitely be a pain for many developers. Even with the best IDE. And we want to ease things...
I let others give their opinions.

**Comment by pankajagrawal16** [comment]
> I agree the shorter the better. Ideally it would be: `Logger`, `Tracer` and ...?  and `@Log`, `@Trace`, `@LogMetrics` 
> 
> But having two Loggers will definitely be a pain for many developers. Even with the best IDE. And we want to ease things...
> 
> I let others give their opinions.

We should definitely not force qualified imports especially when we know our logger utility is based on log4j which will conflict with that name. I agree with that.

I personally don't dislike having Lambda in the name and making it explicit for lambda, although I don't have strong opinions if we skip using lambda in the name.

How about usinh EnhancedLogger and @EnhancedLogging ? And agreeing on rest like above?

Thoughts?

**Comment by dgomesbr** [comment]
What are the thoughts on implementing  #80? So no logging problems at all

**Comment by pankajagrawal16** [comment]
> What are the thoughts on implementing #80? So no logging problems at all

I replied there separately. I think this a separate issue which should be discussed and evaluated anyways together with community feedback before we go GA with this.

**Comment by stevehouel** [comment]
Agreed with @dgomesbr,  IDE will resolve those naming context for us. So with that we will keep the naming conventions that we have on other packages and have a kind of consistency in our naming conventions. 

So +1 for @Logger, @Tracer

**Comment by msailes** [comment]
GA is approaching, and we'd like to take this chance to change any API names before we move to v1.0.0

These are our current names. We'd like to remove `Powertools`, keep the names short and memorable. We also want to make sure not to clash with any other class we know we'll be expecting. For example we don't want to name our core logging annotation `@Logger` as it will clash with Log4j.

| Current | Proposed |
| :--- | :--- |
PowertoolsTracing | Tracing
PowertoolsLogging | Logging
PowertoolsMetrics | Metrics
LargeMessageHandler | SqsLargeMessage
SqsBatchProcessor | SqsBatch
Validation | Validation |

**Comment by dgomesbr** [comment]
Thanks Mark! Can we decide between verbs and nouns? 

* Tracer, Logger, Metrics, looks better.
* Validation / Validators+Validator, don't have an well formed idea on it
* SqsLargeMessage -> LargeMessageParser ?
* BatchMessageProcessor ?

**Comment by pankajagrawal16** [comment]
> Thanks Mark! Can we decide between verbs and nouns?
> 
> * Tracer, Logger, Metrics, looks better.
> * Validation / Validators+Validator, don't have an well formed idea on it
> * SqsLargeMessage -> LargeMessageParser ?
> * BatchMessageProcessor ?

Hey @dgomesbr , we don't really want to force user to do a fully qualified imports by naming annotation as `@Logger`, especially when we know utility depends on log4j2 Logger which will always force user for have qualifies import on one or the other.  Its also hard then to follow always follow a specific pattern

**Comment by pankajagrawal16** [comment]
> GA is approaching, and we'd like to take this chance to change any API names before we move to v1.0.0
> 
> These are our current names. We'd like to remove `Powertools`, keep the names short and memorable. We also want to make sure not to clash with any other class we know we'll be expecting. For example we don't want to name our core logging annotation `@Logger` as it will clash with Log4j.
> 
> Current	Proposed
> PowertoolsTracing	Tracing
> PowertoolsLogging	Logging
> PowertoolsMetrics	Metrics
> LargeMessageHandler	SqsLargeMessage
> SqsBatchProcessor	SqsBatch
> Validation	Validation

Guess then we will go ahead an update as per proposed namings.
